### ISSUE TITLE
Enable Drawing in Presentation View

### ISSUE THREAD

**Issue by stettberger** [pdfpc/pdfpc#518]
The feature wish description:

In drawing mode (keybinding 2, 3), I want to be able to draw within the _presentation_ window like I'm able to draw in the Presenter.main_view. 

Currently, any mouse action, regardless of the mode, in the presentation window goes to the next slide. I think, changing this behavior would make the semantic of drawing mode more consistent.

**Comment by fnevgeny** [comment]
But *why* do you want it?

**Comment by stettberger** [comment]
I use a wacom tablet for drawing. And it is much easier to pin the Wacom drawing surface onto the presentation screen to directly draw there.

**Comment by fnevgeny** [comment]
> I use a wacom tablet for drawing. And it is much easier to pin the Wacom drawing surface onto the presentation screen to directly draw there.

Sorry, but I have to repeat the question: why? Perhaps it's obvious for a user of the Wacom device, but I don't have one. BTW, did you see #324?

**Comment by stettberger** [comment]
The Wacom device is an absolute input device. This means that one coordinate on the tablet maps to a fixed position on the screen. The Wacom driver allows it to map the full width of the tablet to the full width of an entire screen. By drawing into the presentation window it is much easier to hit the correct spot on the slide.

**Comment by fnevgeny** [comment]
OK, so #324 is now implemented. Does it solve your problem then?

**Comment by danilocesar** [comment]
> OK, so #324 is now implemented. Does it solve your problem then?

Another user using an absolute device here. #324 is not perfect, but it helps a lot.

**Comment by fnevgeny** [comment]
@danilocesar just to understand, what is really "not perfect" - that you are limited to ~90% of the full tablet area or that the resulting rendering is not pixel-for-pixel correct?

**Comment by danilocesar** [comment]
I was talking about the first thing (being limited to 90% of the tablet) - but way better than ~50% as before #324.
The second one is very minor, really not a problem from my own use-case. I can imagine that, if you are doing something more complex, the pixel-per-pixel thing can be a problem.

Just to be sure about your pixel-for-pixel correctness comment (sorry, I'm very new to pdfpc): when you're talking about the pixel-for-pixel correctness, you're talking about the 1280px constrains of the drawing area. Is that right?

**Comment by fnevgeny** [comment]
> when you're talking about the pixel-for-pixel correctness, you're talking about the 1280px constrains of the drawing area. Is that right?

Nope. The drawings are rendered using the canvas size of the presenter, rescaled to the actual dimensions of the presentation screen. In practice, this is rarely a serious issue since the presenter screen (a laptop with at least full HDMI, not to mention HiDPI/Retina displays) typically exceeds the physical overhead resolution (if you run pdfpc in the windowed mode and squeeze the presenter to a small size, the problem becomes more evident). Still, rescaling by a fractional number degrades the quality. I think ultimately, we should keep the drawings in a vector format internally and render to both the presenter and presentation canvases separately. BTW, this would also allow for a more elegant implementation of [undo](https://github.com/pdfpc/pdfpc/issues/548).

**Comment by danilocesar** [comment]
It makes a lot of sense.

**Comment by jirilebl** [comment]
Perhaps I can add another "why".  I like pdfpc for presentations but currently I cannot use it for online teaching.  That is, drawing is unusable if I am teaching by sharing screen, or to create videos by capturing screen or window.  pympress does work this way (drawing on presentation window) so that's what I am currently planning on using.

**Comment by fnevgeny** [comment]
@jirilebl This is a reasonable "why" :)
